Source/DafnyCore/Verifier/BoogieGenerator.ExpressionWellformed.cs

```diff
@@ -1537,7 +1537,7 @@ private record ReadsCheckDelayer(ExpressionTranslator etran, Function selfCallsA
         // assume (forall i0,i1,i2,... :: { nw[i0,i1,i2,...] }
         //            0 <= i0 < ... && ... ==> nw[i0,i1,i2,...] == init.requires(i0,i1,i2,...));
         var ai = ReadHeap(tok, etran.HeapExpr, nw, GetArrayIndexFieldName(tok, indices));
-        var ai_prime = UnboxIfBoxed(ai, elementType);
+        var ai_prime = TrType(elementType) == predef.BoxType ? ai : ApplyUnbox(tok, ai, TrType(elementType));
```

Comment:
I liked having only one function here, even if it was used only once. Would you consider re-adding it?